opensquirrel/exporter/writer.py

```diff
@@ -4,35 +4,35 @@
 class _WriterImpl(SquirrelIRVisitor):
     number_of_significant_digits = 8
 
-    def __init__(self, number_of_qubits, qubit_register_name):
+    def __init__(self, number_of_qubits: int, qubit_register_name: str) -> None:
         self.qubit_register_name = qubit_register_name
         self.output = f"""version 3.0\n\nqubit[{number_of_qubits}] {qubit_register_name}\n\n"""
 
-    def visit_qubit(self, qubit: Qubit):
+    def visit_qubit(self, qubit: Qubit) -> str:
         return f"{self.qubit_register_name}[{qubit.index}]"
 
-    def visit_int(self, i: Int):
+    def visit_int(self, i: Int) -> str:
         return f"{i.value}"
 
-    def visit_float(self, f: Float):
+    def visit_float(self, f: Float) -> str:
         return f"{f.value:.{self.number_of_significant_digits}}"
 
-    def visit_measure(self, measure: Measure):
+    def visit_measure(self, measure: Measure) -> None:
         self.output += f"{measure.name} {measure.arguments[0].accept(self)}\n"
 
-    def visit_gate(self, gate: Gate):
+    def visit_gate(self, gate: Gate) -> None:
         if gate.is_anonymous:
             self.output += "<anonymous-gate>\n"
             return
 
         formatted_args = (arg.accept(self) for arg in gate.arguments)
```

Comment:
This line is broken with the following error message:
```terminal
Item "None" of "tuple[Expression, ...] | None" has no attribute "__iter__" (not iterable)  [union-attr]
```

Reply:
I got quite a similar error this afternoon due to using `list` instead of `List`.